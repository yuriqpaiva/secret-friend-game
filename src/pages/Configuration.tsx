import React from 'react';
import Card from '../components/Card';
import Form from '../components/Form';
import ParticipantList from '../components/Form/ParticipantList';
import Footer from '../components/Layout/Footer';

const Configuration: React.FC = () => {
  return (
    <Card>
      <section>
        <h2>Vamos come√ßar!</h2>
        <Form />
        <ParticipantList />
        <Footer />
      </section>
    </Card>
  );
};

export default Configuration;
