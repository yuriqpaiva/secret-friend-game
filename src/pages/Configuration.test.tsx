import { render } from '@testing-library/react';
import React from 'react';
import { RecoilRoot } from 'recoil';
import Configuration from './Configuration';

const mockNavigation = jest.fn();

jest.mock('react-router-dom', () => ({
 useNavigate: () => mockNavigation,
}));

describe('Configuration page', () => {
  test('Must be rendered correctly', () => {
    const { container } = render(
      <RecoilRoot>
        <Configuration />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
