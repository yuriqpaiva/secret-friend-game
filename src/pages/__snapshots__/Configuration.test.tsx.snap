// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration page Must be rendered correctly 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Vamos come√ßar!
      </h2>
      <form>
        <div
          class="form-container"
        >
          <input
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Adicionar
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="footer-configurations"
      >
        <button
          class="footer-configurations__button"
          disabled=""
        >
          Iniciar brincadeira
        </button>
        <img
          alt="Sacolas de compras"
          src="sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
